lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -455,6 +455,24 @@ SupplementaryOutputPathsComputer::computeOutputPathsForOneInput(
       file_types::TY_ModuleTrace, "",
       defaultSupplementaryOutputPathExcludingExtension);
 
+  // We piggy-back on the loadedModuleTracePath to decide (1) whether
+  // to emit the fine module Trace file, and (2) where to emit the fine module
+  // trace file if the path isn't explicitly given by
+  // SWIFT_COMPILER_OBJC_MESSAGE_TRACE_PATH.
+  // FIXME: we probably need to move this to a frontend argument.
+  llvm::SmallString<128> FineModuleTracePath;
+  if (!loadedModuleTracePath.empty()) {
+    if (const char *P = ::getenv("SWIFT_COMPILER_OBJC_MESSAGE_TRACE_PATH")) {
```

Comment:
Can you remove the OBJC from this and use FINE_GRAINED instead?

Reply:
Will fix!